include/ibamr/StaggeredStokesPhysicalBoundaryHelper.h

```diff
@@ -92,16 +92,26 @@ class StaggeredStokesPhysicalBoundaryHelper : public IBTK::StaggeredPhysicalBoun
      * \brief At open boundaries, set normal velocity ghost cell values to
      * enforce the discrete divergence-free condition in the ghost cell abutting
      * the physical boundary.
+     *
+     * \note If enforce_at_all_phy_bdrys = true, then the discrete divergence-free condition will be enforced
+     * at all boundaries.
      */
-    void enforceDivergenceFreeConditionAtBoundary(int u_data_idx, int coarsest_ln = -1, int finest_ln = -1) const;
+    void enforceDivergenceFreeConditionAtBoundary(int u_data_idx,
+                                                  int coarsest_ln = -1,
+                                                  int finest_ln = -1,
+                                                  bool enforce_at_all_phy_bdrys = false) const;
```

Comment:
This last flag seems redundant. Isn't that what this function is supposed to do anyway?

Reply:
The flag enforces div free condition in the ghost cells. The original IBAMR routine does it for just traction BC. We find it necessary for all BCs to get 2nd order convergence and stability for multiphase flows.

Reply:
Can you update the comment for the method not to start with "At open boundaries...". Maybe it could say something like:

"Set normal velocity ghost cell values to enforce ... . The default behavior is to set these values only in cells adjacent to boundary locations where normal traction conditions are imposed."

Finally, is there any possibility that someone may want only to do this at locations where normal velocity boundary conditions are imposed? Seems unlikely, but if so, maybe it would be better to use a bitmask that controls where these values are set that defaults to a value named `NORMAL_TRACTION` but that could be or'ed with another value named `NORMAL_VELOCITY`? (It seems preferable to me to have named quantities instead of boolean values, to make it more clear what is being controlled, but the boolean also is OK here.)

Reply:
Hmm, I can update the comment. Do you want to take care of this named variables and bit mask thing? I don't have a good understanding of how bit mask thing works😬

Reply:
I just pushed some code that is meant to do this. Take a look, and feel free to change the tag names.

Reply:
Thanks! Looks good to me. Nishant will run suite-of-tests with latest commit to confirm order of accuracy and stability for high density ratio flows. Once passed you can merge it.